scanner/config.go

```diff
@@ -26,16 +26,25 @@ const (
 	detected
 )
 
+const (
+	optionsFailedTag        = "options_failed"
+	configsFailedTag        = "configs_failed"
+	detectPlatformFailedTag = "detect_platform_failed"
+	noPlatformDetectedTag   = "no_platform_detected"
+)
+
 type scannerOutput struct {
 	status status
 
 	// can always be set
 	// warnings returned by DetectPlatform(), Options()
-	warnings models.Warnings
+	warnings                   models.Warnings
+	warningsWithRecommendation []models.ErrorWithRecommendations
```

Comment:
If models.Warnings is an array maybe []models.ErrorWithRecommendations should become 

models.ErrorsWithRecommendations too.